databricks/koalas/frame.py

```diff
@@ -7247,6 +7248,40 @@ def append(
 
         return cast(DataFrame, concat([self, other], ignore_index=ignore_index))
 
+    def combine_first(self, other: "DataFrame") -> "DataFrame":
+        if isinstance(other, ks.Series):
+            other = other.to_frame()
+
+        update_columns = list(
+            set(self._internal.column_labels).intersection(set(other._internal.column_labels))
+        )
+
+        update_sdf = combine_frames(self, other)._internal.resolved_copy.spark_frame
```

Comment:
How about create `combined_df` first and get the `updated_sdf` after then so that we can use another items of `combined_df` ??

like

```python
        combined_df = combine_frames(self, other)
        update_sdf = combined_df._internal.resolved_copy.spark_frame
        ...
        spark_columns = combined_df._internal.spark_columns
        update_sdf.select(spark_columns, ...)
```